packages/utils/src/constants/network.ts

```diff
@@ -0,0 +1,91 @@
+// Gossip constants
+
+/**
+ * The maximum number of slots during which an attestation can be propagated.
+ */
+export const ATTESTATION_PROPAGATION_SLOT_RANGE = 32;
+
+//  Request/Response constants
+
+export enum RespStatus {
+  /**
+   * A normal response follows, with contents matching the expected message schema and encoding specified in the request
+   */
+  SUCCESS = 0,
+  /**
+   * The contents of the request are semantically invalid, or the payload is malformed,
+   * or could not be understood. The response payload adheres to the ErrorMessage schema
+   */
+  INVALID_REQUEST = 1,
+  /**
+   * The responder encountered an error while processing the request. The response payload adheres to the ErrorMessage schema
+   */
+  SERVER_ERROR = 2,
+  /**
+   * The responder does not have requested resource.  The response payload adheres to the ErrorMessage schema (described below). Note: This response code is only valid as a response to BlocksByRange
+   */
+  RESOURCE_UNAVAILABLE = 3,
+  /**
+   * Our node does not have bandwidth to serve requests due to either per-peer quota or total quota.
+   */
+  RATE_LIMITED = 139,
+}
+
+export type RpcResponseStatusError = Exclude<RespStatus, RespStatus.SUCCESS>;
+
+/** The maximum allowed size of uncompressed gossip messages. */
+export const GOSSIP_MAX_SIZE = 2 ** 20;
+export const GOSSIP_MAX_SIZE_BELLATRIX = 10 * GOSSIP_MAX_SIZE;
+/** The maximum allowed size of uncompressed req/resp chunked responses. */
+export const MAX_CHUNK_SIZE = 2 ** 20;
+export const MAX_CHUNK_SIZE_BELLATRIX = 10 * MAX_CHUNK_SIZE;
+/** The maximum time to wait for first byte of request response (time-to-first-byte). */
+export const TTFB_TIMEOUT = 5 * 1000; // 5 sec
+/** The maximum time for complete response transfer. */
+export const RESP_TIMEOUT = 10 * 1000; // 10 sec
+/** Non-spec timeout from sending request until write stream closed by responder */
+export const REQUEST_TIMEOUT = 5 * 1000; // 5 sec
+/** Non-spec timeout from dialing protocol until stream opened */
+export const DIAL_TIMEOUT = 5 * 1000; // 5 sec
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const timeoutOptions = {TTFB_TIMEOUT, RESP_TIMEOUT, REQUEST_TIMEOUT, DIAL_TIMEOUT};
+
+export enum GoodByeReasonCode {
+  CLIENT_SHUTDOWN = 1,
+  IRRELEVANT_NETWORK = 2,
+  ERROR = 3,
+  TOO_MANY_PEERS = 129,
+  SCORE_TOO_LOW = 250,
+  BANNED = 251,
+}
+
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const GOODBYE_KNOWN_CODES: Record<string, string> = {
+  0: "Unknown",
+
+  // spec-defined codes
+  1: "Client shutdown",
+  2: "Irrelevant network",
+  3: "Internal fault/error",
+
+  // Teku-defined codes
+  128: "Unable to verify network",
+
+  // Lighthouse-defined codes
+  129: "Client has too many peers",
+  250: "Peer score too low",
+  251: "Peer banned this node",
+};
+
+/** Until js-libp2p types its events */
+export enum Libp2pEvent {
+  peerConnect = "peer:connect",
+  peerDisconnect = "peer:disconnect",
```

Comment:
By no means the utils package has to be aware of all these deep implementation details of the beacon-node. Please remove and keep the PR diff minimal